Modules/Loadable/Markups/MRML/vtkMRMLMarkupsROINode.cxx

```diff
@@ -137,6 +138,121 @@ void vtkMRMLMarkupsROINode::CreateDefaultDisplayNodes()
   this->SetAndObserveDisplayNodeID(dispNode->GetID());
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsROINode::GetRASBounds(double bounds[6])
+{
+  vtkBoundingBox box;
+  box.GetBounds(bounds);
+
+  if (this->ROIType == ROITypeBox || this->ROIType == ROITypeBoundingBox)
+    {
+    double xAxisWorld[3] = { 0.0, 0.0, 0.0 };
+    this->GetXAxisWorld(xAxisWorld);
+    double yAxisWorld[3] = { 0.0, 0.0, 0.0 };
+    this->GetYAxisWorld(yAxisWorld);
+    double zAxisWorld[3] = { 0.0, 0.0, 0.0 };
+    this->GetZAxisWorld(zAxisWorld);
+    double centerWorld[3] = { 0.0, 0.0, 0.0 };
+    this->GetCenterWorld(centerWorld);
+
+    vtkNew<vtkPoints> boxCornerPointsWorld;
+    this->GenerateBoxCornerPoints(boxCornerPointsWorld, xAxisWorld, yAxisWorld, zAxisWorld, centerWorld, this->Size);
+    for (int i = 0; i < boxCornerPointsWorld->GetNumberOfPoints(); ++i)
+      {
+      box.AddPoint(boxCornerPointsWorld->GetPoint(i));
+      }
+    }
+
+  box.GetBounds(bounds);
+}
+
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsROINode::GetBounds(double bounds[6])
```

Comment:
It would be nice to reduce code duplication between internal implementation of GetBounds and GetBoundsRAS (for example, more code could be moved into the shared GenerateBoxCornerPoints method).

Reply:
Changed GenerateBoxCornerPoints -> GenerateBoxBounds. All bounds calculation is now done within GenerateBoxBounds.